compiler/src/iree/compiler/Codegen/LLVMGPU/Utils/PrefetchSharedMemoryCopy.cpp

```diff
@@ -176,9 +176,54 @@ class LoopPrefetcher {
     });
   }
 
+  void getValueDependenciesForIf(scf::IfOp ifOp,
+                                 DenseSet<Operation *> &readDependencies,
+                                 DenseSet<Operation *> &writeDependencies) {
+    // scf.if with results should be supported directly through the usual
+    // handling so bail-out here in that case
+    if (ifOp->getNumResults() != 0)
+      return;
+    bool hasGlobalRead = false;
+    bool hasSharedWrite = false;
+    // Else region not yet supported.
+    if (!ifOp.getElseRegion().empty()) {
+      return;
+    }
+    ifOp->walk([&](Operation *op) {
+      if (auto readOp = dyn_cast<vector::TransferReadOp>(op)) {
+        auto sourceType = dyn_cast<MemRefType>(readOp.getBase().getType());
+        if (hasGlobalMemoryAddressSpace(sourceType)) {
+          hasGlobalRead = true;
+        }
```

Comment:
Can you add a test case to cover when the address space is `#amdgpu.address_space<fat_raw_buffer>`? It should be recognized as a global read but I'm not really sure.

Reply:
The `hasGlobalMemoryAddressSpace` function checks this and returns true for fat_raw_buffer, so adding address_sapce will really be testing that utility. I think we shouldnt add it as this pass shouldnt care about backend specific address spaces and I dont want too many tests. Let me know if you think differently.